### ISSUE TITLE
Extend TD003 to allow "Jira-style" issue IDs on the same line

### ISSUE THREAD

**Issue by simonpercivall** [astral-sh/ruff#20809]
### Summary

Rule: https://docs.astral.sh/ruff/rules/missing-todo-link/#missing-todo-link-td003

Extension to https://github.com/astral-sh/ruff/issues/8061, where this was changed for "GitHub-style" issue numbers, to allow them on the same line as the `TODO` comment.

### Current behavior
For issue _numbers_, this is allowed:
`# TODO: Move this part to the other place #6877`

but this is not:
`# TODO: Move this part to the other place RFFU-6877`

### Outcome
TD003 should accept issue IDs of the form `<PREFIX>-<NUMBER>` on the same line as the TODO comment, i.e.:

`# TODO: Move this part to the other place RFFU-6877`

**Comment by MichaReiser** [comment]
I'm not that familiar with Jira style IDs but I think this is already supported:

```py
# TODO: this link has no issue
# PROJ-123
```

https://play.ruff.rs/10e254a1-e087-4163-9ffe-32977ec0ec99

Unless your IDs use a different format

**Comment by simonpercivall** [comment]
Right, but that's not on the same line.

I edited the description to add an example. Sorry for being unclear in the description.

**Comment by MichaReiser** [comment]
Oh right, I missed that part. 

I'd be fine with this change. We added same-line support in https://github.com/astral-sh/ruff/pull/15519 and followed VS Code's set of rules. But I don't see why the set of supported codes between same and own line should differ (it only seems confusing). @dylwil3 do you remember why we decided to use two different regex sets?

**Comment by dylwil3** [comment]
I believe it was just to make the minimal change necessary in order to resolve the issue - probably I was more hesitant to make changes in general back then 😄 I am certainly comfortable expanding the allowed forms

**Comment by snailUlitka** [comment]
Can I work on this issue?

**Comment by ntBre** [comment]
It looks like @danparizher already has an open PR, but thank you for asking!

**Comment by snailUlitka** [comment]
Oh, I missed that, my bad! Thanks for pointing it out :)
